### PR TITLE
PLANET-3058: Fix master-theme phpcs errors

### PR DESCRIPTION
**Pull request by pablocubico**
Ref: https://jira.greenpeace.org/browse/PLANET-3058

Note: the Master Site class from functions.php had to be moved to its own file to follow the PHPCS standard and class naming convention.

### CODE REVIEW

File: classes/class-p4-campaigns.php [hunk 1]
```diff
@@ -27,14 +44,14 @@ public function __construct() {
 		 * Class hooks.
 		 */
 		private function hooks() {
-			add_action( 'post_tag_add_form_fields',              array( $this, 'add_taxonomy_form_fields' ) );
-			add_action( 'post_tag_edit_form_fields',             array( $this, 'add_taxonomy_form_fields' ) );
-			add_action( 'create_post_tag',                       array( $this, 'save_taxonomy_meta' ) );
-			add_action( 'edit_post_tag',                         array( $this, 'save_taxonomy_meta' ) );
-			add_action( 'admin_enqueue_scripts',                 array( $this, 'enqueue_admin_assets' ) );
+			add_action( 'post_tag_add_form_fields', array( $this, 'add_taxonomy_form_fields' ) );
```
Comment (kirdia) [inline classes/class-p4-campaigns.php]: Suggestion: Would be nice if you can transform these to short array notation.
eg. `[ $this, 'add_taxonomy_form_fields'  ]`

File: classes/class-p4-control-panel.php [hunk 2]
```diff
@@ -22,15 +27,15 @@ public function __construct() {
 		public function hooks() {
 			// Display the Control Panel only to Administrators.
 			if ( current_user_can( 'manage_options' ) || current_user_can( 'editor' ) ) {
-				add_action( 'wp_dashboard_setup',              array( $this, 'add_dashboard_widgets' ), 9 );
-				add_action( 'wp_ajax_flush_cache',             array( $this, 'flush_cache' ) );
-				add_action( 'wp_ajax_check_cache',             array( $this, 'check_cache' ) );
-				add_action( 'admin_enqueue_scripts',           array( $this, 'enqueue_admin_assets' ) );
+				add_action( 'wp_dashboard_setup', array( $this, 'add_dashboard_widgets' ), 9 );
```
Comment (kirdia) [inline classes/class-p4-control-panel.php]: Same here for short array notation.

File: classes/class-p4-custom-taxonomy.php [hunk 3]
```diff
@@ -20,14 +25,14 @@ public function __construct() {
 		 * Register actions for WordPress hooks and filters.
 		 */
 		private function hooks() {
-			add_action( 'init',                               array( $this, 'register_taxonomy' ), 2 );
-			add_action( 'created_term',                       array( $this, 'trigger_rewrite_rules' ), 10, 3 );
-			add_action( 'edited_term',                        array( $this, 'trigger_rewrite_rules' ), 10, 3 );
-			add_action( 'delete_term',                        array( $this, 'trigger_rewrite_rules' ), 10, 3 );
-			add_action( 'save_post',                          array( $this, 'save_taxonomy_page_type' ) , 10, 2 );
+			add_action( 'init', array( $this, 'register_taxonomy' ), 2 );
+			add_action( 'created_term', array( $this, 'trigger_rewrite_rules' ), 10, 3 );
```
Comment (kirdia) [inline classes/class-p4-custom-taxonomy.php]: Same here for short array notation.

File: page-templates/evergreen.php [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<?php /* Template Name: Evergreen Page */
```
Comment (kirdia) [inline page-templates/evergreen.php]: This is needed for WordPress to pick up templates.
https://developer.wordpress.org/themes/template-files-section/page-template-files/

An alternative could be this, 
```
/**
 * Template Name: Evergreen Page
 * The template for displaying evergreen pages.
```

Comment (pablocubico) [reply page-templates/evergreen.php]: Ugh! Didn't know that, I'll change it.

**Comment by pablocubico** [comment]
@kirdia done!

**Comment by kirdia** [comment]
Looks good overall, well done!
Just two things.

1. Please remove composer lock from repo.
2. I would prefer to keep the tests filenames as they were, as they would be consistent with plugin-blocks and also `wp scaffold` generates them with that naming. Although, I don't think there is a naming convention for this. What do you think?

Here is how you can ignore the sniff errors for the test classes filenames.
https://github.com/greenpeace/planet4-plugin-blocks/blob/develop/phpcs.xml#L25

File: phpcs.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+<ruleset name="WordPress Coding Standards for Plugins">
+	<description>Generally-applicable sniffs for WordPress plugins</description>
+
+	<rule ref="WordPress-Core" />
+	<rule ref="WordPress-Docs" />
+
+	<!-- Whitelist p4 test classes for filename sniffs. -->
+	<rule ref="WordPress.Files.FileName">
+		<properties>
+			<property name="custom_test_class_whitelist" type="array" value="P4_TestCase,P4MasterThemeTest,OpenGraphTest,CustomTaxonomyTest,CustomPermalinksTest,CategoryPageTest"/>
```
Comment (kirdia) [inline phpcs.xml]: No need to add all the class names. Just the class that they extend. This should be enough.
	

	<rule ref="WordPress.Files.FileName">
		<properties>
			<property name="custom_test_class_whitelist" type="array" value="P4_UnitTestCase"/>
		</properties>
	</rule>`

[custom unit test classes](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#custom-unit-test-classes)

Also there is already a phpcs configuration file in repo (phpcs.xml.dist), please add this rule there

Comment (pablocubico) [reply phpcs.xml]: Ahhh, that's much better! Will do.

**Comment by pablocubico** [comment]
@kirdia Done! Updated the branch with the latest release, fixed some conflicts with the custom taxonomy class.

**Comment by comzeradd** [comment]
@pablocubico It seems there is still a conflict :D

**Comment by kirdia** [comment]
@pablocubico Looks good! Still a conflict there because #701 was merged

File: phpcs.xml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
```
Comment (kirdia) [inline phpcs.xml]: This file is still added in the repo, please remove it also

**Comment by pablocubico** [comment]
@kirdia Duplicated file removed, rebased to latest release, fixed last sniffs. 💪

**Comment by kirdia** [comment]
Great work @pablocubico!! Thanx you! :+1:

**Pull request merged**